src/inverse/SIA2D/gradient.jl

```diff
@@ -242,19 +317,30 @@ function SIA2D_grad_batch!(θ, simulation::Inversion)
             λ₁ = zero(H[end])
 
             # Include contribution of loss from last step since this is not accounted for in the discrete callback
-            if tspan[2] ∈ t_ref
+            if tspan[2] ∈ tstops
```

Comment:
Is there any way with the new logic with the timestepts to remove the need from this code and include all together?